pages/multiplexed-samples/[uuid].js

```diff
@@ -0,0 +1,198 @@
+              <MultiplexedTable
+                multiplexedSamples={multiplexedSample.multiplexed_samples}
+              />
+            </>
+          )}
+          {multiplexedSample.file_sets?.length > 0 && (
+            <>
+              <DataAreaTitle>File Sets</DataAreaTitle>
+              <FileSetTable fileSets={multiplexedSample.file_sets} />
+            </>
+          )}
+          {multiplexedSample.modifications?.length > 0 && (
+            <>
+              <DataAreaTitle>Modifications</DataAreaTitle>
+              <ModificationsTable
+                modifications={multiplexedSample.modifications}
+              />
+            </>
+          )}
+          {biomarkers.length > 0 && (
+            <>
+              <DataAreaTitle>Biomarkers</DataAreaTitle>
+              <BiomarkerTable biomarkers={biomarkers} />
+            </>
+          )}
+          {treatments.length > 0 && (
+            <>
+              <DataAreaTitle>Treatments</DataAreaTitle>
+              <TreatmentTable treatments={treatments} />
+            </>
+          )}
+          {documents.length > 0 && (
+            <>
+              <DataAreaTitle>Documents</DataAreaTitle>
+              <DocumentTable documents={documents} />
+            </>
+          )}
+          <Attribution attribution={attribution} />
+        </JsonDisplay>
+      </EditableItem>
+    </>
+  );
+}
+
+MultiplexedSample.propTypes = {
+  // MultiplexedSample-cell sample to display
+  multiplexedSample: PropTypes.object.isRequired,
+  // Documents associated with the sample
+  documents: PropTypes.arrayOf(PropTypes.object).isRequired,
+  // Donors associated with the sample
+  sources: PropTypes.arrayOf(PropTypes.object),
+  // Treatments associated with the sample
+  treatments: PropTypes.arrayOf(PropTypes.object).isRequired,
+  // Biomarkers of the sample
+  biomarkers: PropTypes.arrayOf(PropTypes.object).isRequired,
+  // Attribution for this sample
+  attribution: PropTypes.object,
+  // Is the format JSON?
+  isJson: PropTypes.bool.isRequired,
+};
+
+export async function getServerSideProps({ params, req, query }) {
+  const isJson = isJsonFormat(query);
+  const request = new FetchRequest({ cookie: req.headers.cookie });
+  const multiplexedSample = await request.getObject(
+    `/multiplexed-samples/${params.uuid}/`
+  );
+  if (FetchRequest.isResponseSuccess(multiplexedSample)) {
+    const documents = multiplexedSample.documents
+      ? await requestDocuments(multiplexedSample.documents, request)
+      : [];
+    const donors = multiplexedSample.donors
+      ? await requestDonors(multiplexedSample.donors, request)
+      : [];
+    let sources = [];
+    if (multiplexedSample.sources?.length > 0) {
+      const sourcePaths = multiplexedSample.sources.map(
+        (source) => source["@id"]
+      );
+      sources = await request.getMultipleObjects(sourcePaths, null, {
+        filterErrors: true,
+      });
```

Comment:
Generally we’re moving to `getMultipleObjectsBulk` which can get many objects in just one or a couple requests instead of as many requests as there are objects. But because `getMultipleObjectsBulk` additionally requires a list of object properties to retrieve, we’ve been generally putting these requests behind all those `requestBlah` functions so that we only need to list those properties in one place, not everywhere we need thm. So you could add a `requestSources()` function to common-requests.js that follows the pattern of the other functions in there, adding an array of the properties you need. It looks like you only need `name` and `url`.

Reply:
done